### PR TITLE
Added setting default time during conversion from date to time storing formats

### PR DESCRIPTION
**Pull request by gotfrydoh**

### CODE REVIEW

**Review by jradlica — APPROVED** [review]

File: src/test/java/net/kaczmarzyk/e2e/converter/TimestampE2eTest.java [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+import java.time.LocalDateTime;
+
+import static net.kaczmarzyk.spring.data.jpa.CustomerBuilder.customer;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+public class TimestampE2eTest extends E2eTestBase {
+
+	@Controller
```
Comment (tkaczmarzyk) [inline src/test/java/net/kaczmarzyk/e2e/converter/TimestampE2eTest.java:44]: fix formatting

File: src/test/java/net/kaczmarzyk/e2e/converter/TimestampE2eTest.java [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+
+import static net.kaczmarzyk.spring.data.jpa.CustomerBuilder.customer;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+public class TimestampE2eTest extends E2eTestBase {
```
Comment (tkaczmarzyk) [inline src/test/java/net/kaczmarzyk/e2e/converter/TimestampE2eTest.java:42]: consider adding javadoc with `@author`

File: src/test/java/net/kaczmarzyk/e2e/converter/TimestampE2eTest.java [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+
+        @RequestMapping(value = "/customers", params = "lastSeenBefore")
+        @ResponseBody
+        public Object findCustomersWithLastSeenBefore_defaultTimestampPattern(
+                @Spec(path="lastSeen", params="lastSeenBefore", spec= LessThan.class) Specification<Customer> spec) {
+
+            return customerRepo.findAll(spec);
+        }
+
+		@RequestMapping(value = "/customers", params = "lastSeenBefore_customFormat")
```
Comment (tkaczmarzyk) [inline src/test/java/net/kaczmarzyk/e2e/converter/TimestampE2eTest.java:58]: please check formatting of the entire class

File: src/test/java/net/kaczmarzyk/spring/data/jpa/Customer.java [hunk 4]
```diff
@@ -61,6 +62,8 @@ public class Customer {
 
     private LocalDateTime lastOrderTime;
 
+	private Timestamp lastSeen;
```
Comment (tkaczmarzyk) [inline src/test/java/net/kaczmarzyk/spring/data/jpa/Customer.java:65]: formatting

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,8 +1,12 @@
+v2.15.0
+=======
+* added setting default time during conversion from date to time storing formats in `Converter`. Missing time values are filled with zeros (e.g. `LocalDateTime` format specified as `2022-12-14` will be converted to `2022-12-14 00:00`). Similar time setting is implemented for the following formats: `LocalDateTime`, `Timestamp`, `Instant`, `OffsetDateTime`.
```
Comment (tkaczmarzyk) [inline CHANGELOG.md:3]: added support for using datetime formats without time (e.g. `yyyy-MM-dd`) for types that contain time (`LocalDateTime`, `Timestamp`, `Instant`, `OffsetDateTime`). Missing time values are filled with zeros, e.g. when sending `2022-12-14` as `LocalDateTime` parameter, the conversion will result in `2022-12-14 00:00`.

**Review by tkaczmarzyk — CHANGES_REQUESTED** [review]

**Review by tkaczmarzyk — APPROVED** [review]

**Pull request merged**